chimney/src/main/scala/io/scalaland/chimney/internal/compiletime/derivation/transformer/rules/TransformSubtypesRuleModule.scala

```diff
@@ -8,17 +8,10 @@ trait TransformSubtypesRuleModule { this: Definitions & Derivation =>
 
   object TransformSubtypesRule extends Rule {
 
-    def isApplicableTo[From, To](implicit ctx: TransformerContext[From, To]): Boolean = {
-      Type[From] <:< Type[To]
-    }
+    def isApplicableTo[From, To](implicit ctx: TransformerContext[From, To]): Boolean = Type[From] <:< Type[To]
 
-    def apply[From, To](implicit ctx: TransformerContext[From, To]): DerivationResult[DerivedExpr[To]] = {
-      DerivationResult.pure(
-        DerivedExpr.TotalExpr[To](
-          ctx.src.asInstanceOfExpr[To]
-        )
-      )
-    }
+    def apply[From, To](implicit ctx: TransformerContext[From, To]): DerivationResult[DerivedExpr[To]] =
+      DerivationResult.log(s"Matched ${Type.prettyPrint[From]} <:< ${Type.prettyPrint[To]} derivation rule") >>
```

Comment:
Instead of this, how about generic log before calling 'apply', that we're expanding rule of specific name?

Reply:
Depends how we want to log recursive derivations and customization, IMHO ideally each nesting/recursion should result in ins own indentation in logs, so wherever we would put these logs it would be in some place that isn't accidentally skipped when we didn `Rule[From, To](someSubsetOfRules)`.

Reply:
I don't think we should directly call `apply` method of another rules. Even if you do, you would need to maintain nesting/indentation manually.

I believe we need a better/more explicit way to call logic from another rules.

Reply:
Directly not. But we reserve ourselves the right to call something like `Rule(someSubsetOfRules)` somehow filter out some rules in recursive calls. And then it will be nested and annotated.

Reply:
We concluded to remodel Rule interface that enables:
- explicit rules sequence swap in the future, if we ever need it
- continuing expansion, with the current rules sequence, following result of the complex logic (potentially recursive expansion)